src/gateway_api/client/gateway_client_dispatch_request.rs

```diff
@@ -0,0 +1,126 @@
+use crate::prelude::*;
+
+///
+/// Internal
+///
+impl GatewayClient {
+    /// Dispatches an HTTP `POST` request by JSON serializing the specified
+    /// `request` and setting it as the `body` for the network request.
+    pub(crate) async fn post<T, U, V, F>(
+        &self,
+        path: impl AsRef<str>,
+        request: T,
+        map: F,
+    ) -> Result<V, CommonError>
+    where
+        T: Serialize,
+        U: for<'a> Deserialize<'a>,
+        F: Fn(U) -> Result<V, CommonError>,
+    {
+        self.dispatch_network_request(path, NetworkMethod::Post, request, map)
+            .await
+    }
+
+    /// Dispatches an HTTP `POST` request without any `body`.
+    pub(crate) async fn post_empty<U, V, F>(
+        &self,
+        path: impl AsRef<str>,
+        map: F,
+    ) -> Result<V, CommonError>
+    where
+        U: for<'a> Deserialize<'a>,
+        F: Fn(U) -> Result<V, CommonError>,
+    {
+        #[derive(Serialize)]
+        struct EmptyBodyPostRequest {}
+        self.post(path, EmptyBodyPostRequest {}, map).await
+    }
+}
+
+/// An identity mapping function for Result
+pub(crate) const fn res_id<T>(x: T) -> Result<T, CommonError> {
+    std::convert::identity::<Result<T, CommonError>>(Ok(x))
+}
+
+///
+/// Private
+///
+impl GatewayClient {
+    fn model_from_response<U>(
+        &self,
+        response: NetworkResponse,
+    ) -> Result<U, CommonError>
+    where
+        U: for<'a> Deserialize<'a>,
+    {
+        if let 200..=299 = response.status_code {
+            // all good
+        } else {
+            return Err(CommonError::NetworkResponseBadCode);
+        }
+
+        if response.body.is_empty() {
+            return Err(CommonError::NetworkResponseEmptyBody);
+        }
+
+        serde_json::from_slice::<U>(&response.body).map_err(|_| {
+            CommonError::NetworkResponseJSONDeserialize {
+                into_type: std::any::type_name::<U>().to_owned(),
+            }
+        })
+    }
+
+    async fn dispatch_network_request<T, U, V, F>(
```

Comment:
renamed from `make_request` to "dispatch", which I think is a good name... tells FFI side (iOS/Android) to execute the request.